### ISSUE TITLE
lintrunner doesn't work on deleted files

### ISSUE THREAD

**Issue by H-Huang** [suo/lintrunner#14]
This is not an urgent issue, but I deleted a file as part of my changes and `lintrunner -a` fails.

rage:
```
lintrunner rage report:
timestamp: 2022-08-31T13:23:00.243-07:00
args: 'lintrunner' '-a' '-v'
exit code: 1
err msg: Some("Failed to find file while gathering files to lint: /private/home/howardhuang/pytorch-projects/pytorch/torch/csrc/distributed/c10d/OpsImpl.hpp")

========= BEGIN LOGS =========
[2022-08-31T20:23:00Z DEBUG lintrunner] Version: 0.9.2
[2022-08-31T20:23:00Z DEBUG lintrunner] Passed args: Args { inner: ["lintrunner", "-a", "-v"] }
[2022-08-31T20:23:00Z DEBUG lintrunner] Computed args: Args { verbose: 1, config: ".lintrunner.toml", apply_patches: true, paths_cmd: None, paths_from: None, revision: None, merge_base_with: None, skip: None, take: None, output: Default, cmd: None, paths: [], force_color: false, data_path: None, tee_json: None, all_files: false }
[2022-08-31T20:23:00Z DEBUG lintrunner] Current rev: d956dae3dd5d03bc845b62e9972fa1fc81c33f3f
[2022-08-31T20:23:00Z DEBUG lintrunner::lint_config] Found linters: {"MYPY", "NEWLINE", "NOQA", "CIRCLECI", "NATIVEFUNCTIONS", "CALL_ONCE", "PYBIND11_INCLUDE", "FLAKE8", "SPACES", "INCLUDE", "CLANGTIDY", "TABS", "PYPIDEP", "EXEC", "PYBIND11_SPECIALIZATION", "CUBINCLUDE", "WORKFLOWSYNC", "UFMT", "SHELLCHECK", "ACTIONLINT", "RAWCUDA", "MYPYSTRICT", "CLANGFORMAT", "TYPEIGNORE", "ONCE_FLAG", "CMAKE", "TESTOWNERS"}
[2022-08-31T20:23:00Z DEBUG lintrunner::persistent_data] Checking data file '/private/home/howardhuang/.local/share/lintrunner/bead93266017a343c0460d4b7f1e341fd1377ace6b639541fd53a8ced23af967/.lintrunner.toml' to see if config has changed
[2022-08-31T20:23:00Z DEBUG lintrunner] Running linters: ["FLAKE8", "CLANGFORMAT", "MYPY", "MYPYSTRICT", "CLANGTIDY", "TYPEIGNORE", "NOQA", "CIRCLECI", "NATIVEFUNCTIONS", "NEWLINE", "SPACES", "TABS", "INCLUDE", "PYBIND11_INCLUDE", "PYBIND11_SPECIALIZATION", "PYPIDEP", "EXEC", "CUBINCLUDE", "RAWCUDA", "CMAKE", "SHELLCHECK", "ACTIONLINT", "TESTOWNERS", "CALL_ONCE", "ONCE_FLAG", "WORKFLOWSYNC", "UFMT"]
[2022-08-31T20:23:00Z TRACE lintrunner::log_utils] Linting commit diff files: {"torch/csrc/distributed/c10d/OpsImpl.cpp", "torch/csrc/distributed/c10d/OpsImpl.hpp", "test/distributed/test_c10d_gloo.py", "build_variables.bzl", "test/forward_backward_compatibility/check_forward_backward_compatibility.py", "test/distributed/test_c10d_common.py", "test/distributed/test_c10d_nccl.py"}
[2022-08-31T20:23:00Z TRACE lintrunner::log_utils] Linting working tree diff files: {"torch/csrc/distributed/c10d/Ops.cpp", "torch/csrc/distributed/c10d/OpsImpl.cpp", "test/distributed/test_c10d_gloo.py"}
[2022-08-31T20:23:00Z DEBUG lintrunner::persistent_data] Writing run info to /private/home/howardhuang/.local/share/lintrunner/bead93266017a343c0460d4b7f1e341fd1377ace6b639541fd53a8ced23af967/runs/2022-08-31T13:23:00.243-07:00_7f8c328e06d0cb89332e8065321ffcbd903d13f421ea1a9a7c8004f12c742224
```

**Comment by suo** [comment]
Quite strange, this case should be handled. What did your working tree look like at the time?

**Comment by z-a-f** [comment]
Same error here. Here is the flow that I had that caused the error:

```bash
$ echo "from dummy import dummy" > test.py
$ git add .; git commit -m "test test"
$ lintrunner  # This works
$ git mv test.py dummy.py
$ lintrunner -av  # This fails with the following
lintrunner -av
[2022-09-16T22:34:41Z DEBUG lintrunner] Version: 0.9.2
[2022-09-16T22:34:41Z DEBUG lintrunner] Passed args: Args { inner: ["lintrunner", "-av"] }
[2022-09-16T22:34:41Z DEBUG lintrunner] Computed args: Args { verbose: 1, config: ".lintrunner.toml", apply_patches: true, paths_cmd: None, paths_from: None, revision: None, merge_base_with: None, skip: None, take: None, output: Default, cmd: None, paths: [], force_color: false, data_path: None, tee_json: None, all_files: false }
[2022-09-16T22:34:41Z DEBUG lintrunner] Current rev: 6a7e85b1192e938fac2eafdf2b6f6ead380ea3e7
[2022-09-16T22:34:41Z DEBUG lintrunner::lint_config] Found linters: {"PYBIND11_SPECIALIZATION", "CUBINCLUDE", "CMAKE", "TABS", "TESTOWNERS", "CALL_ONCE", "CLANGTIDY", "ONCE_FLAG", "WORKFLOWSYNC", "RAWCUDA", "UFMT", "PYPIDEP", "NOQA", "TYPEIGNORE", "SHELLCHECK", "SPACES", "NEWLINE", "INCLUDE", "ACTIONLINT", "EXEC", "NATIVEFUNCTIONS", "MYPYSTRICT", "PYBIND11_INCLUDE", "FLAKE8", "CIRCLECI", "MYPY", "CLANGFORMAT"}
[2022-09-16T22:34:41Z DEBUG lintrunner::persistent_data] Checking data file '/data/home/zaf/.local/share/lintrunner/77d66ca9521d64f608a437157d209dbeb074c941acd695e7eb1bc258315dc21a/.lintrunner.toml' to see if config has changed
[2022-09-16T22:34:41Z DEBUG lintrunner] Running linters: ["FLAKE8", "CLANGFORMAT", "MYPY", "MYPYSTRICT", "CLANGTIDY", "TYPEIGNORE", "NOQA", "CIRCLECI", "NATIVEFUNCTIONS", "NEWLINE", "SPACES", "TABS", "INCLUDE", "PYBIND11_INCLUDE", "PYBIND11_SPECIALIZATION", "PYPIDEP", "EXEC", "CUBINCLUDE", "RAWCUDA", "CMAKE", "SHELLCHECK", "ACTIONLINT", "TESTOWNERS", "CALL_ONCE", "ONCE_FLAG", "WORKFLOWSYNC", "UFMT"]
[2022-09-16T22:34:42Z DEBUG lintrunner::persistent_data] Writing run info to /data/home/zaf/.local/share/lintrunner/77d66ca9521d64f608a437157d209dbeb074c941acd695e7eb1bc258315dc21a/runs/2022-09-16T22:34:41.691Z_4cb9c61a98f36ca9a4fc63caa78dcb4db86021c87c8c7436fdf1ebe77c6050d2
error:        Failed to find file while gathering files to lint: test.py
caused_by:             No such file or directory (os error 2)
```

However, if I commit the change, the error is gone:

```bash
$ git status
On branch dummy_branch
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        renamed:    test.py -> dummy.py
$ git add -A; git commit --amend
$ lintrunner  # Works fine
```

**Comment by suo** [comment]
Thanks @z-a-f and @H-Huang for the repros. This has been fixed in [v0.9.3](https://github.com/suo/lintrunner/blob/main/CHANGELOG.md#bug-fixes), so if you run `pip install --upgrade` you should get the fix.
